disperser/common/v2/blobstore/dynamo_store.go

```diff
@@ -0,0 +1,283 @@
+package blobstore
+
+import (
+	"context"
+	"fmt"
+	"strconv"
+	"strings"
+	"time"
+
+	commondynamodb "github.com/Layr-Labs/eigenda/common/aws/dynamodb"
+	core "github.com/Layr-Labs/eigenda/core/v2"
+	"github.com/Layr-Labs/eigenda/disperser"
+	v2 "github.com/Layr-Labs/eigenda/disperser/common/v2"
+	"github.com/Layr-Labs/eigensdk-go/logging"
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go-v2/feature/dynamodb/attributevalue"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb"
+	"github.com/aws/aws-sdk-go-v2/service/dynamodb/types"
+)
+
+const (
+	StatusIndexName            = "StatusIndex"
+	OperatorDispersalIndexName = "OperatorDispersalIndex"
+	OperatorResponseIndexName  = "OperatorResponseIndex"
+
+	blobKeyPrefix     = "BlobKey#"
+	dispersalIDPrefix = "DispersalID#"
+	blobMetadataSK    = "BlobMetadata"
+	certificateSK     = "Certificate"
+)
+
+// blobMetadataStore is a blob metadata storage backed by DynamoDB
+type blobMetadataStore struct {
+	dynamoDBClient *commondynamodb.Client
+	logger         logging.Logger
+	tableName      string
+	ttl            time.Duration
+}
+
+var _ BlobMetadataStore = (*blobMetadataStore)(nil)
+
+func NewBlobMetadataStore(dynamoDBClient *commondynamodb.Client, logger logging.Logger, tableName string, ttl time.Duration) BlobMetadataStore {
+	logger.Debugf("creating blob metadata store v2 with table %s with TTL: %s", tableName, ttl)
+	return &blobMetadataStore{
+		dynamoDBClient: dynamoDBClient,
+		logger:         logger.With("component", "BlobMetadataStoreV2"),
+		tableName:      tableName,
+		ttl:            ttl,
+	}
+}
+
+func (s *blobMetadataStore) PutBlobMetadata(ctx context.Context, blobMetadata *v2.BlobMetadata) error {
+	item, err := MarshalBlobMetadata(blobMetadata)
+	if err != nil {
+		return err
+	}
+
+	return s.dynamoDBClient.PutItem(ctx, s.tableName, item)
+}
+
+func (s *blobMetadataStore) GetBlobMetadata(ctx context.Context, blobKey core.BlobKey) (*v2.BlobMetadata, error) {
+	item, err := s.dynamoDBClient.GetItem(ctx, s.tableName, map[string]types.AttributeValue{
+		"PK": &types.AttributeValueMemberS{
+			Value: blobKeyPrefix + blobKey.Hex(),
+		},
+		"SK": &types.AttributeValueMemberS{
+			Value: blobMetadataSK,
+		},
+	})
+
+	if item == nil {
+		return nil, fmt.Errorf("%w: metadata not found for key %s", disperser.ErrMetadataNotFound, blobKey.Hex())
+	}
+
+	if err != nil {
+		return nil, err
+	}
+
+	metadata, err := UnmarshalBlobMetadata(item)
+	if err != nil {
+		return nil, err
+	}
+
+	return metadata, nil
+}
+
+// GetBlobMetadataByStatus returns all the metadata with the given status
+// Because this function scans the entire index, it should only be used for status with a limited number of items.
+func (s *blobMetadataStore) GetBlobMetadataByStatus(ctx context.Context, status v2.BlobStatus) ([]*v2.BlobMetadata, error) {
```

Comment:
Would there ever be a need for a batched version of this?

Reply:
What do you mean by a batched version?